### ISSUE TITLE
The OS magnifier can't follow the text input cursor in editor

### ISSUE THREAD

**Issue by JiayangShen** [microsoft/vscode#7394]
- VSCode Version:1.1.1
- OS Version:windows7

Steps to Reproduce:
1.  Press Win + R shortcut key to open the "Run Dialog", and input "magnify" to open the magnifier tool in the windows system. Then setting the magnifier to follow mouse pointer, follow keyboard focus, and follow text input cursor. 
2.  Open vscode, new a blank document, input some long text. You will find that the magnifier can't follow the text input cursor in editor.

**Thanks for this great code editor! But my eyesight is bad, magnifier is needed. I know the input cursor is just a simulated one using html+css+js, is there any way to make it can be followed by the OS magnifier when input in the editor?**

**Comment by JiayangShen** [comment]
@alexandrudima 
Hi, alexandrudima, I'm glad to saw that you labeled this issue. It's very important to me. Till now, modern code editors, like sublime text and atom, and vscode, all of them have this accessibility issue. I like vscode so much that I really hope this issue can be solved as quickly as possible. Thanks!

**Comment by alexandrudima** [comment]
A possible solution would be positioning the `<textarea>` always with the cursor. Assigning to September to investigate if this would be possible.

**Comment by alexandrudima** [comment]
Unfortunately, I can reproduce this in a Chromium native `<input>` box such as the `ctrl+p` input box. Properly configured, the magnifier does reveal the cursor location when doing e.g. composition in CJK languages, but not otherwise. Please consider filing an issue against Chromium

**Comment by Deon588** [comment]
This has finally been fixed in:

https://bugs.chromium.org/p/chromium/issues/detail?id=142133&can=2&start=0&num=100&q=&colspec=ID%20Pri%20M%20Stars%20ReleaseBlock%20Component%20Status%20Owner%20Summary%20OS%20Modified&groupby=&sort=

@alexandrudima
 
What is the next step?  Any idea When we can expect the  new chromium to be added to vsCode?

**Comment by alexandrudima** [comment]
VS Code currently uses Electron 1.7.9, which uses Chromium 58.0.3029.110.

We update Electron regularly, and we will pick up the fix eventually. 

I am sorry I cannot give an estimate on how long it will take until this happens.

**Comment by DeonNel1** [comment]
@@alexandrudima it's been almost 4 months and stis still isn't working.  Any idea which version of chromium the latest vsCode is using?  Just trying to determine whether this has actually been fixed in chromium or if it's just a case of electron not using the version of Chromium which had the fix implemented yet?  Anyone else struggling with this a quick workaround/temporary partial fix is to not use the windows magnifier in the editor and just zoom in while using the editor by pressing ctrl++

**Comment by alexandrudima** [comment]
@DeonNel1 

VS Code currently is using `Electron 1.7.9`, which comes with `Chromium 58.0.3029.110`.

We are in the process of adopting `Electron v2.0.0-beta.1`, which comes with `Chromium 59.0.3071.115`, but we have hit a few blocking Electron regressions.

**Comment by JiayangShen** [comment]
Very much hope that this issue can be solved...

**Comment by daniahl** [comment]
Agreed.
I was under the impression that a more recent Electron/Chromium version would fix this, but I open dev tools and run process.versions, and see chrome: "66.0.3359.181" electron:
"3.1.6"
Meanwhile, in Atom, where focus tracking works, I do the same thing and get chrome: "61.0.3163.100"
electron: "2.0.18"

Maybe there is some configuration parameter that has to be set to enable the necessary events to be emitted?

**Comment by renatorroliveira** [comment]
This issue happens in so many editors that is hard to say if updating libs will fix it, probably not. I feel like this is something that he editor needs to support actively to work.

It`s just like KDE's KMag tool which I have been asking for this feature and the docked mode feature for 10 years by now. Here at least everyone is trying to solve the issue

**Comment by isidorn** [comment]
Thanks for @coryj627 for providing more details and a great video showcasing this problem.
After investigation with @alexandrudima we found the workaround for https://github.com/Microsoft/vscode/issues/26730 was causing the magnifier not following the cursosr issue - because the hidden text area would always be at the top left.
Since the workaround is no longer needed (due to other fixes in NVDA) we have removed the workaround and both the magnifier works nice on my machine and https://github.com/Microsoft/vscode/issues/26730 is not regressing.

Thanks all for feedback on this issue and please keep the issues comming.

fyi @zersiax
